### ISSUE TITLE
eslint package not respecting version configuration

### ISSUE THREAD

**Issue by mikejgray** [projen/projen#3951]
I've noticed some conflicts with eslint@^9, so to work around that for now I've tried to set it to ^8 via the `devDeps` property in my project (AwsCdkTypeScriptApp). However, it ignores my version specification unless I use `project.package.addDevDeps('eslint@^8');`

I suspect something in the package code is overriding the `devDeps` value, which is why the method works but setting the property does not.

**Comment by mrgrain** [comment]
No, that's just the way to do it.

**Comment by mrgrain** [comment]
Note to myself: There is a way that component can define dependency requirements that allow users to overwrite them with `devDeps`. Not sure if we should use this here. But if not, we should probably throw or warn instead of just silently fail.

tl;dr Components declaring dependencies is currently not transparent enough.

**Comment by DaveArchetype** [comment]
observation: after I added `project.package.addDevDeps('eslint@^8');` to each of the packages inside projenrc, eslint did use version 8. it is as if something overwrites the eslint version in the main package.json if any of the other packages use a newer eslint version
